tests/test_neps_api/test_api.py

```diff
@@ -0,0 +1,204 @@
+import logging
+import os
+import runpy
+from pathlib import Path
+
+import pytest
+import yaml
+
+
+# To change the working directly into the tmp_path when testing function
+@pytest.fixture(autouse=True)
+def use_tmpdir(tmp_path, request):
+    os.chdir(tmp_path)
+    yield
+    os.chdir(request.config.invocation_dir)
+
+
+# https://stackoverflow.com/a/59745629
+# Fail tests if there is a logging.error
+@pytest.fixture(autouse=True)
+def no_logs_gte_error(caplog):
+    yield
+    errors = [
+        record
+        for record in caplog.get_records("call")
+        if record.levelno >= logging.ERROR
+    ]
+    assert not errors
+
+
+# Expected outcomes of the optimizer YAML according to different cases of neps.run
+# all based on the examples in tests/test_neps_api/examples_test_api.py
+expected_dicts = {
+    "priorband_bo_user_decided": {
+        "searcher_name": "priorband_bo",
+        "searcher_alg": "priorband",
+        "searcher_selection_source": "Default_Searcher-User_Choice",
+        "searcher_modified_arguments": {
+            "initial_design_size": 5,
+        },
+    },
+    "bo_user_decided": {
+        "searcher_name": "bayesian_optimization",
+        "searcher_alg": "bayesian_optimization",
+        "searcher_selection_source": "Default_Searcher-User_Choice",
+        "searcher_modified_arguments": {
+            "surrogate_model": "ComprehensiveGPHierarchy",
+            "surrogate_model_args": {
+                "graph_kernels": [
+                    "WeisfeilerLehman",
+                    "WeisfeilerLehman",
+                    "WeisfeilerLehman",
+                    "WeisfeilerLehman",
+                    "WeisfeilerLehman",
+                ],
+                "hp_kernels": [],
+                "verbose": False,
+                "hierarchy_consider": [0, 1, 2, 3],
+                "d_graph_features": 0,
+                "vectorial_features": None,
+            },
+        },
+    },
+    "priorband_neps_decided": {
+        "searcher_name": "priorband",
+        "searcher_alg": "priorband",
+        "searcher_selection_source": "Default_Searcher-NePS_Decision_Tree",
```

Comment:
Seeing these here, seems like they should be `ClassVar`'s on the objects themselves?

Reply:
I changed these to have solution_yamls, which is easier to then locate and add for future tests I assumed.